provider/cmd/pulumi-resource-random/schema.json

```diff
@@ -227,151 +183,162 @@
-                    "type": "integer",
-                    "description": "Minimum number of special characters in the result.\n"
+                    "type": "number",
+                    "description": "Minimum number of special characters in the result. Default value is `0`.\n"
                 },
                 "minUpper": {
-                    "type": "integer",
-                    "description": "Minimum number of uppercase alphabet characters in the result.\n"
+                    "type": "number",
+                    "description": "Minimum number of uppercase alphabet characters in the result. Default value is `0`.\n"
                 },
                 "number": {
                     "type": "boolean",
-                    "description": "Include numeric characters in the result.\n"
+                    "description": "Include numeric characters in the result. Default value is `true`. **NOTE**: This is deprecated, use `numeric` instead.\n",
+                    "deprecationMessage": "**NOTE**: This is deprecated, use `numeric` instead."
+                },
+                "numeric": {
+                    "type": "boolean",
+                    "description": "Include numeric characters in the result. Default value is `true`.\n"
                 },
                 "overrideSpecial": {
                     "type": "string",
-                    "description": "Supply your own list of special characters to use for string generation. This overrides the default character list in\nthe special argument. The `special` argument must still be set to true for any overwritten characters to be used in\ngeneration.\n"
+                    "description": "Supply your own list of special characters to use for string generation.  This overrides the default character list in the special argument.  The `special` argument must still be set to true for any overwritten characters to be used in generation.\n"
                 },
                 "result": {
                     "type": "string",
-                    "description": "The generated random string.\n"
+                    "description": "The generated random string.\n",
+                    "secret": true
                 },
                 "special": {
                     "type": "boolean",
-                    "description": "Include special characters in the result. These are `!@#$%\u0026*()-_=+[]{}\u003c\u003e:?`\n"
+                    "description": "Include special characters in the result. These are `!@#$%\u0026*()-_=+[]{}\u003c\u003e:?`. Default value is `true`.\n"
                 },
                 "upper": {
                     "type": "boolean",
-                    "description": "Include uppercase alphabet characters in the result.\n"
+                    "description": "Include uppercase alphabet characters in the result. Default value is `true`.\n"
                 }
             },
             "required": [
+                "bcryptHash",
```

Comment:
Ok something is perhaps wrong here? https://github.com/hashicorp/terraform-provider-random/blob/main/internal/provider/resource_password.go#L702 

these are properties specified as Computed: true, Sensitive: true. Why are they now listed under required? need to track as a bug.

Reply:
Some good news investigating this unexpected diff. This could plausibly be by-design. The facts are:

- not marked Computed before:
https://github.com/pulumi/terraform-provider-random/blob/2ed10370a61b5ad8a25f02ad9952052b39b9c7a6/internal/provider/string.go#L180

- marked Computed after:
https://github.com/terraform-providers/terraform-provider-random/blob/e361b1c0647513bf7fc308caa3ecdc3a0bc965f8/internal/provider/resource_string.go#L464

- there is code in TF Bridge that intentionally considers Computed properties required when interpreting them in output position:
https://github.com/pulumi/pulumi-terraform-bridge/blob/ba8306077beace9d280a6c736952ed6cd2b4e7a2/pkg/tfgen/generate.go#L523